rllib/algorithms/algorithm.py

```diff
@@ -2399,16 +2606,31 @@ def _run_one_evaluation(
         Returns:
             The results dict from the evaluation call.
         """
-        eval_results = {"evaluation": {}}
+        eval_results = {
```

Comment:
What about the other eval results like `NUM_AGENT_STEPS_SAMPLED_THIS_ITER`? Would it be cleaner to make this an np.nan default dict?

Reply:
I'm not sure, this would also create weird arbitrary values, like:
`eval_results["evaluation"]["test_xyz"] = np.nan`, no?

I think we are just adding these three basic keys here such that our tuned_examples will work.
We only ever guarantee (if `always_attach_evaluation_results=True`) that eval results will be there once we actually did run a complete evaluation, not before that.